response_operations_ui/controllers/party_controller.py

```diff
@@ -125,6 +127,44 @@ def search_respondent_by_email(email):
     return response.json()
 
 
+def search_respondents(first_name, last_name, email_address, page=0):
+    url = _get_search_respondents_url(first_name=first_name,
+                                      last_name=last_name,
+                                      email_address=email_address,
+                                      page=page)
+    response = requests.get(url, auth=app.config['PARTY_AUTH'])
+    if response.status_code != 200:
+        raise SearchRespondentsException(response,
+                                         first_name=first_name,
+                                         last_name=last_name,
+                                         email_address=email_address,
+                                         page=page)
+    data = response.json()
+
+    return data
+
+
+def _get_search_respondents_url(**kwargs):
+    url = f'{app.config["PARTY_URL"]}/party-api/v1/respondents?'
+
+    # Dict provides translation of variable names in response ops to url param names in party
+    kw_to_url_params = {
+        'first_name': 'firstName',
+        'last_name': 'lastName',
+        'email_address': 'emailAddress',
+        'page': 'page'
+    }
+
+    for resops_name, party_name in kw_to_url_params.items():
+        arg = kwargs.get(resops_name)
+        if arg:
+            url += f'{party_name}={arg}&'
```

Comment:
This looks vulnerable to an injection attack to me

Reply:
That depends on whether the service the url is calling is sanitising it's inputs - I'm told it is.

Either way, I can sanitise this, but the service will remain open to attach itself.

Reply:
Function removed now anyway.